File: core/src/main/kotlin/com/wafflestudio/snu4t/lectures/data/Lecture.kt
package com.wafflestudio.snu4t.lectures.data

import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document
import org.springframework.data.mongodb.core.mapping.Field

@Document("lectures")
class Lecture(
        @Id
        var id: String? = null,
        @Field("academic_year")
        var academicYear: String?,
        var category: String?,
        @Field("class_time")
        var classTimeText: String?,
        @Field("real_class_time")
        var realClassTimeText: String?,
        @Field("class_time_json")
        var classTime: List<ClassTime>,
        @Field("class_time_mask")
        var classTimeMask: List<Int>,
        var classification: String?,
        var credit: Int,
        var department: String?,
        var instructor: String?,
        @Field("lecture_number")
        var lectureNumber: String?,
        var quota: Int?,
        var remark: String?,
        var semester: Int?,
        var year: Int?,
        @Field("course_number")
        var courseNumber: String?,
        @Field("course_title")
        var courseTitle: String?,
)

File: src/core/lecture/model/Lecture.ts
import TimePlace from '@app/core/timetable/model/TimePlace';

export default interface Lecture {
    _id?: string,
    classification: string,                           // 교과 구분
    department: string,                               // 학부
    academic_year: string,                            // 학년
    course_title: string,   // 과목명
    credit: number,                                   // 학점
    class_time: string,
    real_class_time: string,
    class_time_json: TimePlace[],
    class_time_mask: number[],
    instructor: string,                               // 강사
    quota: number,                                    // 정원
    freshmanQuota?: number,                            // 신입생정원
    remark: string,                                   // 비고
    category: string,
}
